### PR TITLE
TEN-1380 / 24.10.0 / Add SFTP password login groups and password login enabled test cases

### PR DESCRIPTION
**Pull request by jordan-mcgillivray005**

### CODE REVIEW

**Comment by jordan-mcgillivray005** [comment]
image

File: helper/cli.py [hunk 1]
```diff
@@ -34,18 +39,62 @@ def __init__(self, command: str, wsl: bool = True):
...
+            print("@@@ STDOUT: " + self.stdout)
+            print("@@@ STDERR: " + self.stderr)
+            print("@@@ STATUS: " + str(self.status))
         # This was causing failures with commands that are meant to fail such as on SSHCOM.remove_all_test_files()
         # when no files are not found. Commenting for now to assess effect.
         # assert self.status
 
 
+class SFTP_Client:
+    def __init__(self, hostname, username, password, port=22):
```
Comment (markbainix) [inline helper/cli.py:52]: are we putting `-> None` on these methods?

File: test_cases/webui/sftp/test_sftp_controlled_access.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        4. Verify the SFTP connection
+        """
+        self.setup_password_login_allowed_test(True)
+        assert SFTP.assert_sftp_access('accessuser', 'test') is True
+        self.setup_password_login_allowed_test(False)
+        assert SFTP.assert_sftp_access('accessuser', 'test') is False
+        self.setup_password_login_allowed_test(True)
+        NAV.navigate_to_system_settings_services()
+        SYS.click_edit_button_by_servicename('SSH')
+        while COM.is_visible('//mat-chip-row//*[@name="cancel"]'):
```
Comment (ericbsd) [inline test_cases/webui/sftp/test_sftp_controlled_access.py:138]: Should we expose xpath at this level?

There are multiple like this in the file.

File: helper/cli.py [hunk 3]
```diff
@@ -34,18 +39,62 @@ def __init__(self, command: str, wsl: bool = True):
...
+            raise Exception(err)
+        finally:
+            print(f"Connected to {self.hostname} as {self.username}.")
+
+    def disconnect(self):
+        """Closes the sftp connection"""
+        self.connection.close()
+        print(f"Disconnected from host {self.hostname}")
+
+    def listdir(self, remote_path):
```
Comment (markbainix) [inline helper/cli.py:84]: should this have a "return" line in the doc block?

File: keywords/ssh/sftp.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+class SFTP:
+    @classmethod
+    def assert_sftp_access(cls, username: str = private_config['USERNAME'], password: str = private_config['PASSWORD']) -> bool:
+        """
+        This method verifies the given user is able to access the system via SFTP connection.
+
+        :param username: is the username of the user
+        :param password: is the password of the user
+
```
Comment (markbainix) [inline keywords/ssh/sftp.py:15]: missing `return` line in doc block

File: test_cases/webui/sftp/test_sftp_controlled_access.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        2. Add one user to the group.
+        3. Enable the SSH service and add the created group as a Password Login Group.
+        4. Verify the user in the group is able to establish an SFTP connection.
+        5. Verify the user not in the group is not able to establish an SFTP connection.
+        6. Verify that admin cannot establish an SFTP connection without password login enabled checked.
+        """
+        NAV.navigate_to_system_settings_services()
+        SYS.click_edit_button_by_servicename('SSH')
+        while COM.is_visible('//mat-chip-row//*[@name="cancel"]'):
+            COM.delete_pill('//mat-chip-row')
```
Comment (ericbsd) [inline test_cases/webui/sftp/test_sftp_controlled_access.py:102]: Could this become a method?

File: helper/cli.py (hunk 1 above)
Comment (jordan-mcgillivray005) [reply helper/cli.py:52]: We dont have any on the other cli methods

File: test_cases/webui/sftp/test_sftp_controlled_access.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        COM.set_checkbox('passwordauth')
+        COM.set_input_field('', 'access_group', True, True)
+        COM.click_save_button_and_wait_for_right_panel()
+        SYS.click_edit_button_by_servicename('SSH')
+        assert COM.is_visible(xpaths.common_xpaths.any_pill("", 'access_group')) is True
```
Comment (ericbsd) [inline test_cases/webui/sftp/test_sftp_controlled_access.py:107]: Should this xpath be in the test file?

Comment (jordan-mcgillivray005) [reply test_cases/webui/sftp/test_sftp_controlled_access.py:107]: Ideally no but it is a waste of time to make a one line method

File: test_cases/webui/sftp/test_sftp_controlled_access.py (hunk 5 above)
Comment (markbainix) [reply test_cases/webui/sftp/test_sftp_controlled_access.py]: agreed. something like `delete_all_pill()` and depending on parameters could be used elsewhere

Comment (jordan-mcgillivray005) [reply test_cases/webui/sftp/test_sftp_controlled_access.py:102]: Made a method. Retesting now

File: helper/cli.py (hunk 1 above)
Comment (ericbsd) [reply helper/cli.py:52]: __init__ does not return anything, it is an initiator, and return doesn't work in __init__.

File: test_cases/webui/sftp/test_sftp_controlled_access.py (hunk 6 above)
Comment (ericbsd) [reply test_cases/webui/sftp/test_sftp_controlled_access.py:104]: If it is only on one test, that should be good, but if other tests start to do the same thing, it should become a method.

Comment (jordan-mcgillivray005) [reply test_cases/webui/sftp/test_sftp_controlled_access.py:104]: I agree

**Review by ericbsd — APPROVED** [review]

File: helper/cli.py (hunk 3 above)
Comment (jordan-mcgillivray005) [reply helper/cli.py:84]: Added

**Review by markbainix — APPROVED** [review]

**Pull request merged**